### PR TITLE
[Accelerator] Do not use NNPA preprocessor to avoid exposing accelerator code

### PR DESCRIPTION
**Pull request by tungld**
This patch is to remove NNPA preprocessor so that accelerator code is not exposed outside its folder.

`InitAccelerators` function is modified to return `bool` instead of `void`. It returns True if there are accelerators initialized. Otherwise, False.

`prepareAccelerator` function is separated into two functions `getOrLoadDialects` and `addPasses` to exactly reflect the function's content.

`OptimizationLevel` option is moved to a common place, that is `OMOption`, to be shared between `onnx-mlir` and `onnx-mlir-opt`, which avoids duplicated declaration.

Signed-off-by: Tung D. Le <tung@jp.ibm.com>

### CODE REVIEW

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
@jenkins-droid test this please

File: src/Compiler/CompilerUtils.cpp [hunk 1]
```diff
@@ -126,14 +126,6 @@ static llvm::cl::opt<std::string> march("march",
     llvm::cl::value_desc("Target a specific architecture type"),
     llvm::cl::cat(OnnxMlirOptions), llvm::cl::ValueRequired);
 
-static llvm::cl::opt<OptLevel> OptimizationLevel(
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:129]: I think we should keep all options that are exposed to the user together. `onnx --help` should show them all under the same category (`OnnxMlirOptions`).  Is my understanding that in `OMOptions.cpp` we should have options that  "provide fine control on passes".  Perhaps you can make this option non-static and expose it to the accelerator via an extern declaration?

File: src/Support/OMOptions.cpp [hunk 2]
```diff
@@ -45,3 +47,11 @@ llvm::cl::opt<std::string> acceleratorTarget("maccel",
     llvm::cl::desc("Specify an accelerator to generate code for\n"
                    "\"NONE\" or \"\" for no accelerator\n"),
     llvm::cl::init(""), llvm::cl::cat(OMPassOptions));
+
+llvm::cl::opt<OptLevel> OptimizationLevel(
```
Comment (etiotto) [inline src/Support/OMOptions.cpp:51]: See my other comment. I think we should keep user facing options together and and use extern declarations to expose it to the accelerator code.  Options that are user facing should show in the --help output under the same category.

File: src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp [hunk 3]
```diff
@@ -86,13 +75,16 @@ void scanAndSetOptLevel(int argc, char **argv) {
...
-      OptimizationLevel = (OptLevel)num;
+      OptimizationLevel = (onnx_mlir::OptLevel)num;
       return;
     }
   }
 }
 
 int main(int argc, char **argv) {
+  // Initialize accelerators if they exist.
+  bool hasAccelerator = InitAccelerators();
```
Comment (etiotto) [inline src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp:86]: [nit]: hasAccelerator -> hasAccelerators. (there could be more than one in the future).

File: src/Support/OMOptions.cpp [hunk 4]
```diff
@@ -14,6 +14,8 @@
 
 #include "src/Support/OMOptions.hpp"
 
+using namespace onnx_mlir;
```
Comment (etiotto) [inline src/Support/OMOptions.cpp:17]: Reminder to update copyright year as well from 2021 to 2022.

File: src/Compiler/CompilerUtils.cpp [hunk 5]
```diff
@@ -1071,7 +1062,8 @@ int compileModule(mlir::OwningOpRef<ModuleOp> &module,
     // for (auto accel : *accTargets) {
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:1062]: Should prob. cleanup commented out code.

**Review by etiotto — COMMENTED** [review]
Is great we are removing the need to #ifdef  out the accelerator code. So I like the direction very much. 

We have an issue with the organization of compiler options. We need to expose compiler options to the main driver `onnx-mlir` as well as the tool `onnx-mlir-opt` and some of the accelerator code.  Would be nice to keep all options in a common library, I think we might need to pull the compiler options declaration into a separate header file, and "package" their definitions into a thin library that can be then linked by both `onnx-mlir` and `onnx-mlir-opt` (and perhaps by NNPACompilerUtils as well).

File: src/Compiler/CompilerUtils.cpp (hunk 1 above)
Comment (tungld) [reply src/Compiler/CompilerUtils.cpp:129]: I agree, options are now quite scattered. I think it requires more work to reorganize options, so I have no plan to do it in this patch.

File: src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp (hunk 3 above)
Comment (tungld) [reply src/Tools/onnx-mlir-opt/onnx-mlir-opt.cpp]: Done. Thanks!

File: src/Support/OMOptions.cpp (hunk 4 above)
Comment (tungld) [reply src/Support/OMOptions.cpp:17]: Done. Thanks!

**Review by caoimhinuibrian — CHANGES_REQUESTED** [review]
I'd like to have a meeting to talk about this

**Review by caoimhinuibrian — APPROVED** [review]
LGTM .... sorry, I misunderstood what you were doing

**Pull request merged**